sematic/api_client.py

```diff
@@ -97,25 +146,111 @@ def _put(endpoint, json_payload) -> Any:
     return response.json()
 
 
-class APIConnectionError(requests.exceptions.ConnectionError):
-    pass
+def _validate_server_compatibility(
+    tries: int = 5, seconds_between_tries: int = 10, use_cached: bool = True
+):
+    """Check that the client is compatible with the server.
+
+    Raises an error if the server and client are incompatible, or if this can't be
+    verified.
+    """
+    global _validated_client_version
+    if _validated_client_version and use_cached:
+        return
+
+    retriable_errors = (
+        ConnectionError,
+        ServerError,
+    )
+    retry(
+        exceptions=retriable_errors,
+        tries=tries,
+        delay=seconds_between_tries,
+    )(_validate_server_compatibility_no_catch)()
+
+
+def _validate_server_compatibility_no_catch() -> None:
+    unexpected_server_response_error = IncompatibleClientError(
+        "The Sematic server did not provide information about its version "
+        "in the expected format. It could be that the server is too old to "
+        "provide this information. Consider upgrading your server if possible, "
+        "or reach out to Sematic for support."
+    )
+
+    try:
+        response = _request(
+            method=requests.get,
+            endpoint="/meta/versions",
+            attempt_auth=False,
+            validate_version_compatibility=False,  # to avoid recursion
+            validate_json=True,
+        )
+    except BadRequestError:
+        raise unexpected_server_response_error
+    except ServerError:
+        raise ServerError("The Sematic server is not responsive")
+    except InvalidResponseError:
+        raise unexpected_server_response_error
```

Comment:
```suggestion
except (BadRequestError, InvalidResponseError):
    raise IncompatibleClientError(
        "The Sematic server did not provide information about its version "
        "in the expected format. It could be that the server is too old to "
        "provide this information. Consider upgrading your server if possible, "
        "or reach out to Sematic for support."
    )
except ServerError:
    raise ServerError("The Sematic server is not responsive")
```

Does that make sense?